internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -605,6 +646,79 @@ func (r *AgentServiceConfigReconciler) newAgentRoute(ctx context.Context, log lo
 	return route, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) newHTTPRoute(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig, serviceToExpose string) (client.Object, controllerutil.MutateFn, error) {
+	// In order to create plain http route we need https route to be created first to copy its host
+	httpsRoute := &routev1.Route{}
+	if err := r.Get(ctx, types.NamespacedName{Name: serviceToExpose, Namespace: r.Namespace}, httpsRoute); err != nil {
+		log.WithError(err).Errorf("Failed to get https route for %s service", serviceToExpose)
+		return nil, nil, err
+	}
+	if httpsRoute.Spec.Host == "" {
+		log.Infof("https route for %s service found, but host not yet set", serviceToExpose)
+		return nil, nil, nil
+	}
+
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-ipxe", serviceToExpose),
+			Namespace: r.Namespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: httpsRoute.Spec.Host,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		Path: "/",
+		To: routev1.RouteTargetReference{
+			Kind:   "Service",
+			Name:   serviceToExpose,
+			Weight: &weight,
+		},
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.FromString(fmt.Sprintf("%s-http", serviceToExpose)),
+		},
+		WildcardPolicy: routev1.WildcardPolicyNone,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
+			return err
+		}
+		// Only update what is specified above in routeSpec.
+		// If we update the entire route.Spec with
+		// route.Spec = routeSpec
+		// it would overwrite any existing values for route.Spec.Host
+		route.Spec.To = routeSpec.To
+		route.Spec.Port = routeSpec.Port
+		route.Spec.WildcardPolicy = routeSpec.WildcardPolicy
+		route.Spec.TLS = routeSpec.TLS
+		route.Spec.Path = routeSpec.Path
+		return nil
+	}
+
+	return route, mutateFn, nil
+}
+
+func (r *AgentServiceConfigReconciler) removeHTTPIPXERoute(ctx context.Context, instance *aiv1beta1.AgentServiceConfig, serviceToExpose string) error {
+	route := &routev1.Route{}
+	routeName := fmt.Sprintf("%s-ipxe", serviceToExpose)
+	namespacedName := types.NamespacedName{Name: routeName, Namespace: r.Namespace}
+	if err := r.Get(ctx, namespacedName, route); err == nil {
+		return r.Client.Delete(ctx, &routev1.Route{
```

Comment:
It's a good idea to detect a NotFound error here and return `nil` in that case.

Reply:
Previous `r.Get` would ensure that `r.Client.Delete` cannot return NotFound, isn't it? There might a race there however

Reply:
The Get is just from the client's cache, which exacerbates the potential race that would exist anyway.

Reply:
Right, added an additional check

Reply:
Addressed